wgpu-hal/src/metal/adapter.rs

```diff
@@ -501,6 +501,19 @@ impl super::PrivateCapabilities {
             MTLReadWriteTextureTier::TierNone
         };
 
+        let (max_buffers_per_stage, max_textures_per_stage, max_samplers_per_stage) = if (os_is_mac
+            && rw_texture_tier == MTLReadWriteTextureTier::Tier2)
+            || (family_check && device.supports_family(MTLGPUFamily::Apple6))
+        {
+            (500000 / 2, 500000 / 2, 1024)
```

Comment:
Don't we have the problem of us not using argument buffers?

Reply:
Oh, my mistake, I was only thinking about the maximum value.